tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -116,6 +116,15 @@ spec:
         NUM_MAPPED_COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
         for ((i = 0; i < $NUM_MAPPED_COMPONENTS; i++)) ; do
             component=$(jq -c --argjson i "$i" '.components[$i]' "${SNAPSHOT_SPEC_FILE}")
+
+            # images are required to use sha reference - check this
+            NAME=$(jq -r '.name' <<< "$component")
+            IMAGE_REF=$(jq -r '.containerImage' <<< "$component")
+            if ! [[ "$IMAGE_REF" =~ ^[^:]+@sha256:[0-9a-f]+$ ]] ; then
+              echo "Component ${NAME} contains an invalid containerImage value. sha reference is required: ${IMAGE_REF}"
+              exit 1
+            fi
+
             imageTags=$(jq '.tags // []' <<< "$component")
             git_sha=$(jq -r '.source.git.revision' <<< "$component") # this sets the value to "null" if it doesn't exist
             build_sha=$(jq -r '.containerImage' <<< "$component" | cut -d ':' -f 2)
```

Comment:
Does it make sense to change this to echo $IMAGE_REF | cut -d ':' -f 2

Reply:
Yeah, I thought about it... and wasn't sure if you would like it :D Done.